### PR TITLE
Implement groupTeamDescendantView service

### PR DESCRIPTION
**Pull request by zenovich**
[groupTeamDescendantView](https://docs.google.com/document/d/15Xxmatrs_7my5ioriikL6YgOU4I0sAVcq5ySKinc0Nw/edit#heading=h.ybjtuxidiv0q)

### CODE REVIEW

File: app/api/groups/get_team_descendants.feature [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+      | 22              | 59           | 0       |
+      | 22              | 61           | 0       |
+      | 22              | 63           | 0       |
+      | 22              | 65           | 0       |
+      | 22              | 67           | 0       |
+      | 22              | 69           | 0       |
+
+  Scenario: Get descendant teams
+    Given I am the user with ID "1"
+    # here we fixate time_spent even if it depends on NOW()
```
Comment (smadbe) [inline app/api/groups/get_team_descendants.feature]: I think it is not useful to add it at every usage of the step. Maybe you should just put it in the step definition if you think it is useful.

File: app/api/groups/get_team_descendants.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	}
+
+	var parentsResult []teamDescendantParent
+	service.MustNotBeError(srv.Store.Groups().
+		Select("parent_links.idGroupChild AS idLinkedGroup, groups.ID, groups.sName").
+		Joins(`
+			JOIN groups_groups AS parent_links ON parent_links.idGroupParent = groups.ID AND
+				parent_links.sType = 'direct' AND parent_links.idGroupChild IN (?)`, groupIDs).
+		Joins(`
+			JOIN groups_ancestors AS parent_ancestors ON parent_ancestors.idGroupChild = parent_links.idGroupParent AND
```
Comment (smadbe) [inline app/api/groups/get_team_descendants.go]: ` parent_links.idGroupParent` is actually ` groups.ID`, would be easier to understand.

File: app/api/groups/get_team_descendants.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+	groupIDs := make([]int64, 0, len(result))
+	resultMap := make(map[int64]*teamDescendant, len(result))
+	for index, groupRow := range result {
+		groupIDs = append(groupIDs, groupRow.ID)
+		resultMap[groupRow.ID] = &result[index]
+		result[index].Members = []teamDescendantMember{}
+	}
+
+	var parentsResult []teamDescendantParent
```
Comment (smadbe) [inline app/api/groups/get_team_descendants.go]: If you do it in 3 requests, it assumes you checked it is more efficient than one !?

File: app/api/groups/get_team_descendants.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+type teamDescendant struct {
+	// The team's `groups.ID`
+	// required:true
+	ID int64 `sql:"column:ID" json:"id,string"`
+	// required:true
+	Name string `sql:"column:sName" json:"name"`
+	// required:true
+	Grade int32 `sql:"column:iGrade" json:"grade"`
+
+	// Team's parent groups
```
Comment (smadbe) [inline app/api/groups/get_team_descendants.go]: Add that it is among the input group's descendants.

File: app/api/groups/get_team_descendants.go (hunk 3 above)
Comment (zenovich) [reply app/api/groups/get_team_descendants.go]: optimized queries to make them more efficient + added an index

File: app/api/groups/get_team_descendants.go (hunk 2 above)
Comment (zenovich) [reply app/api/groups/get_team_descendants.go]: fixed

File: app/api/groups/get_team_descendants.feature (hunk 1 above)
Comment (zenovich) [reply app/api/groups/get_team_descendants.feature]: removed DB time freezing (it is not needed here)

File: app/api/groups/get_team_descendants.go (hunk 4 above)
Comment (zenovich) [reply app/api/groups/get_team_descendants.go]: done

**Pull request merged**